presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -61,21 +61,21 @@ public CatalogServer(MetadataManager metadataProvider, SessionPropertyManager se
     public boolean schemaExists(TransactionInfo transactionInfo, SessionRepresentation session, CatalogSchemaName schema)
     {
         transactionManager.tryRegisterTransaction(transactionInfo);
-        return metadataProvider.schemaExists(session.toSession(sessionPropertyManager), schema);
+        return metadata.getMetadataResolver(session.toSession(sessionPropertyManager)).schemaExists(schema);
```

Comment:
At least on this line this does not feel like an improvement. Previously a method was called. Now a single use object is constructed and then a method is called. This isn't per se wrong, but it does suggest this might not be the right abstraction. Why is a session stored in a metadata resolver if it's only used once?

Reply:
@jainxrohit have you synced up with @elharo on the design? Maybe good to provide the progress and see if @elharo has better suggestions.

Reply:
Yeah, I will setup a meeting with Elliotte to describe the intention.